metabase/metabase
# Make editing field settings not rely on the full database metadata

Epic https://github.com/metabase/metabase/issues/30190

Proof-of-concept of editing field settings without full database metadata:
- Hydrates additional fields in field endpoints
- Migrates field settings page to `metabase-lib` and typescript. Make it load all the required data for components to work.
- Adjusts field remapping settings to work with partially loaded data.

How to verify foreign keys:
- Admin -> Data Model -> Orders -> User Id -> Settings
- Make sure it's possible to change every field property from this page
- Make sure it's possible to remap values (Display values -> Use foreign key)
- Make sure it's possible to change set a different foreign key (in Field Type) and remap values to the new table without reloading the page

How to verify custom mapping:
- Admin -> Data Model -> Orders -> Rating -> Settings
- Make sure it's possible to change every field property from this page
- Make sure it's possible to remap values (Display values -> Custom mapping)
- Make sure it's possible to update values in the custom mapping table

<img width="1273" alt="Screenshot 2023-05-01 at 13 29 36" src="https://user-images.githubusercontent.com/8542534/235440903-8d60c5fe-321f-4053-811d-c68ef57e6f18.png">

<img width="735" alt="Screenshot 2023-05-01 at 13 56 19" src="https://user-images.githubusercontent.com/8542534/235443206-bfdc7954-f601-4af3-85d4-af502cab470e.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30423)
